### DISCUSSION TITLE
Define some other enclosing characters that {..} for macro

### DISCUSSION THREAD

**Discussion by ypiel-talend** [verhas/jamal#28] [Q&A] [answered]
Hello,

Is there a way to define some other characters than {...} to all macro ?
For instance JAMAL_MACRO_BEGIN="[" & JAMAL_MACRO_END="]" as environment variable
to have something like:

> [#for $i in (aa bb cc)=* $i
> ]

Regards.

**Comment by verhas** [comment, marked as answer]
Absolutely. You can define macro starting string as a command line parameter, you can define it programmatically when you embed Jamal into an application providing the macro opening and closing string as the contructor argument to the Processor class and last, but not least you can change the macro opening and closing string using the `sep` macro.

Please read the documentation 

https://github.com/verhas/jamal/blob/master/README.adoc#sep

Note that even when you changed the opening and closing strings if you import a file AND the file starts with the characters `{@` then `{` and `}` will be used as macro opening and closing strings inside the included file.

**Comment by verhas** [comment]
It is not recommended to use `[` and `]` as macro opening and closing strings, because the core built-in macros use these characters to enclose options. That way 

```
[#for [evalis] (a,b) in (a,b|c,d)=abraka dabra]
```

will be confusing Jamal as it will not find the macro `evalist`. The same caution is for using `(` and `)`. This is used by the macro `for` and many non-core built-in macros use `(` and `)` to enclose options.
